l10n_it_website_sale_corrispettivi/controllers/main.py

```diff
@@ -8,21 +7,22 @@
 class WebsiteSaleCorrispettivi(WebsiteSale):
 
     def _checkout_form_save(self, mode, checkout, all_values):
-        res = super(WebsiteSaleCorrispettivi, self) \
+        partner_id = super(WebsiteSaleCorrispettivi, self) \
             ._checkout_form_save(mode, checkout, all_values)
         use_invoice = all_values.get('use_invoice')
         # Update order
         order = request.website.sale_get_order()
-        order_values = \
-            self._prepare_corrispettivi_order_values(order, use_invoice)
-        order.write(order_values)
+        order_values = self.sudo()._prepare_corrispettivi_order_values(
```

Comment:
This raises
```
l10n_it_website_sale_corrispettivi/controllers/main.py", line 15, in _checkout_form_save
    order_values = self.sudo()._prepare_corrispettivi_order_values(
AttributeError: 'WebsiteSale (extended by WebsiteSaleCorrispettivi)' object has no attribute 'sudo'
```

Also, using sudo here, in a multi-company environment, if `order.company_id` was empty, this line https://github.com/OCA/l10n-italy/blob/3cc6525fca3a9f1a95c831acaf3e0d208b40bcb5/l10n_it_website_sale_corrispettivi/controllers/main.py#L54 would lead to use the company of admin user, running into undesired behaviour potentially

Reply:
Thanks for your review!
I removed the `sudo()` call for `self` in https://github.com/OCA/l10n-italy/pull/608/commits/79bf9cbee01a1cfa863077c6ec4a870e65163791.
I also changed the *company behavior*: I will always look for the fiscal position having the company of the order (or partner), even if it is `False`.

Reply:
Now I get
```
  File "/home/elbati/workspace/odoo/instances/odoo11/parts/l10n-italy/l10n_it_website_sale_corrispettivi/controllers/main.py", line 16, in _checkout_form_save
    order, use_invoice)
  File "/home/elbati/workspace/odoo/instances/odoo11/parts/l10n-italy/l10n_it_website_sale_corrispettivi/controllers/main.py", line 55, in _prepare_corrispettivi_order_values
    .get_corr_fiscal_pos(company).id
  File "/home/elbati/workspace/odoo/instances/odoo11/parts/l10n-italy/l10n_it_corrispettivi/models/account.py", line 94, in get_corr_fiscal_pos
    ('company_id', '=', company_id.id)], limit=1)
  File "/home/elbati/workspace/odoo/instances/odoo11/parts/odoo/odoo/models.py", line 1480, in search
    res = self._search(args, offset=offset, limit=limit, order=order, count=count)
  File "/home/elbati/workspace/odoo/instances/odoo11/parts/odoo/odoo/models.py", line 3767, in _search
    self.sudo(access_rights_uid or self._uid).check_access_rights('read')
  File "/home/elbati/workspace/odoo/instances/odoo11/parts/odoo/odoo/models.py", line 2862, in check_access_rights
    return self.env['ir.model.access'].check(self._name, operation, raise_exception)
  File "<decorator-gen-20>", line 2, in check
  File "/home/elbati/workspace/odoo/instances/odoo11/parts/odoo/odoo/tools/cache.py", line 89, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/home/elbati/workspace/odoo/instances/odoo11/parts/odoo/odoo/addons/base/ir/ir_model.py", line 1201, in check
    raise AccessError(msg % msg_params)
odoo.exceptions.AccessError: ('Sorry, you are not allowed to access this document. Only users with the following access level are currently allowed to do that:\n- Accounting & Finance/Billing Manager\n\t- Employees/Employee\n\t- Other Extra Rights/Portal\n\n(Document model: account.fiscal.position)', None)
```

Reply:
I tested that in my local environment and fixed that with https://github.com/OCA/l10n-italy/pull/608/commits/5d9bc4087adcad8c77865c5fea915091133a06d8.
The issue was that I was confusing the current user (public user) and the partner that had been created in the checkout: so the public user was not able to read/edit the partner that had just been created.
Anyway, I specified that in the comments.

Again, thanks for the review! :)